### ISSUE TITLE
Avatar is slow when using URI image, suggest cache URI image instead

### ISSUE THREAD

**Issue by rpollock03** [FaridSafi/react-native-gifted-chat#2119]
When passing in a non-local image (i.e. a URI) to the avatar, that image is retrieved for every single message. This is really inefficient and causes a lag in the images, which don't load instantly. For my project, I modified the GiftedAvatar.js file to render a CachedImage component, rather than a regular ReactNative Image.

This CachedImage component is taken from a new file that is similar in functionality to the react-native-expo-cached-image library (https://github.com/wcandillon/react-native-expo-image-cache/blob/master/src/CacheManager.ts).

Basically the URI image is fetched once, cached locally, and then that local cache image is used rather than retrieving an external image. In my project I found this makes for a much faster, smoother use. 

Happy to submit a pull request for this.

**Comment by byanes** [comment]
This would be very helpful

**Comment by byanes** [comment]
Hi! Why was this one closed?

**Comment by Johan-dutoit** [comment]
Everything pre- the v1 release was closed.  If this issue is still present in v1, then worth opening it again.

**Comment by Johan-dutoit** [comment]
I do think it's inefficient, however I feel FastImage should solve this better than implementing custom logic.
